android/app/src/main/java/org/fossasia/openevent/utils/SnackbarUtil.java

```diff
@@ -0,0 +1,52 @@
+package org.fossasia.openevent.utils;
+
+import android.content.Context;
+import android.graphics.PorterDuff;
+import android.support.design.widget.Snackbar;
+import android.support.v4.graphics.ColorUtils;
+import android.view.View;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.SnackbarHelper;
+import org.fossasia.openevent.dbutils.RealmDataRepository;
+
+import static org.fossasia.openevent.listeners.OnBookmarkSelectedListener.CODE_ERROR;
+import static org.fossasia.openevent.listeners.OnBookmarkSelectedListener.CODE_UNDO_ADDED;
+import static org.fossasia.openevent.listeners.OnBookmarkSelectedListener.CODE_UNDO_REMOVED;
+
+public class SnackbarUtil {
+    public static void showSnackbarForBookmark(Context context, View view, int actionCode, SnackbarHelper snackbarHelper) {
+        int toDisplayResource;
+
+        if (actionCode == CODE_UNDO_ADDED)
+            toDisplayResource = R.string.added_bookmark;
+        else if (actionCode == CODE_UNDO_REMOVED)
+            toDisplayResource = R.string.removed_bookmark;
+        else if (actionCode == CODE_ERROR)
+            toDisplayResource = R.string.error_create_notification;
+        else
+            toDisplayResource = R.string.added_bookmark;
+
+        Snackbar snackbar = Snackbar.make(view, toDisplayResource, Snackbar.LENGTH_LONG);
+        if (actionCode == CODE_UNDO_ADDED) {
+            snackbar = snackbar.setAction(R.string.undo, view1 -> {
+                RealmDataRepository.getDefaultInstance().setBookmark(snackbarHelper.getSessionId(), false).subscribe();
+                snackbarHelper.getImageButton().setImageResource(R.drawable.ic_bookmark_white_24dp);
```

Comment:
We only need the snackbar stuff to be handled in this utility class, passing foreign views is bad. Handle what can be handled locally, there itself, create a common static method in `Views` if you need to, but only pass absolutely needed stuff to the listener. I think the only thing needed to pass is the action type and the id of the session.